OneCommunityGlobal/HighestGoodNetworkApp
# Lucas fix tasks another user dashboard

This PR should solve: (PRIORITY LOW) Jae: Fix tasks view when viewing another person’s dashboard

### The idea for the volunteer not be able to see the Owner/Administrator/Core Team data on the dashboard will be implemented on another PR
### In this PR, the volunteer user can see the Onwer/Administrator/Core Team data

<hr>

### Main changes

Previously: When vieweing another user's dashboard, it would show the currently authenticated user's dashboard instead of the viewed user's dashboard;
Now: When the currently authenticated user is seeing another user's dashboard (clicking the green dot on the left of the user, under the Leaderboard), the tasks tab (of the viewed user) will show what the viewed user would see in it's dashboard.


<hr>

### How to test

- Checkout the current branch on frontend
- npm install
- npm run start:local
- Login as various users roles and teams and check other users dashboard
- Go to the tasks tab and see what is shown there
- Check on the user's tasks tab to see if the data that is shown is coherent with the user's dashboard data

### How to check another user's dashboard

- Click on Dashboard
- Go to the Leaderboard
- It should show some users if the logged user is on a team or is an Owner/Administrator/Core Team
- Under the Status theres a green dot
- Click the green dot to check the other user's dashboard
- On the other user's dashboard, see the "Tasks" under the "Tasks and Timelogs"
- The "Tasks" tab should show what the viewed user would see in it's tasks

### How to test (videos) PLEASE SEE THE VIDEOS BEFORE TESTING
### The expected behaviors changed because I've made the changes stated on the item "d" of the task, please review the PR again

What was happening before my pr


https://user-images.githubusercontent.com/92558269/230201921-7777885f-b1c0-4824-ad08-b1eda1212170.mp4



What should you see



https://user-images.githubusercontent.com/92558269/232823611-cdac2e94-0a8f-4d83-a5e7-33fd22cfc282.mp4







<hr>

The changes on the code are commented on the commits.
